offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -369,6 +403,91 @@ export class CkbHandler {
+    }
+    const cellData = await this.parseLockTx(tx);
+    if (!cellData) {
+      return;
+    }
+    const { amount, chain, recipientAddress, recipientCapacity, asset, senderAddress, recipientLockscript } = cellData;
+    const ckbTxHash = tx.hash;
+    const blockNumber = Number(txInfo.info.block_number);
+    const confirmedNumber = currentHeight - blockNumber;
+    const confirmed = confirmedNumber >= ForceBridgeCore.config.ckb.confirmNumber;
+    const confirmStatus = confirmed ? 'confirmed' : 'unconfirmed';
+    const block = await this.ckb.rpc.getBlock(txInfo.tx.txStatus.blockHash!);
+    logger.info(`handle ckb lock tx ${txHash}, confirmed number: ${confirmedNumber}, confirmed: ${confirmed}`);
+    const committeeMultisigLockscript = parseAddress(getOmniLockMultisigAddress());
+    const findCommitteeMultisigOutput = tx.outputs.find(
+      (o) =>
+        o.lock?.codeHash === committeeMultisigLockscript.code_hash &&
+        o.lock.hashType === committeeMultisigLockscript.hash_type,
+    );
+    logger.debug('findCommitteeMultisigOutput:', findCommitteeMultisigOutput);
+    // create new CkbLock record
+    if (records.length === 0 && findCommitteeMultisigOutput) {
+      const capacity = findCommitteeMultisigOutput.capacity;
+      logger.info(
+        `CkbHandler watchLockEvents receiveLog blockHeight:${blockNumber} blockHash:${txInfo.tx.txStatus.blockHash} txHash:${txHash} amount:${capacity} asset:${cellData?.asset} recipientLockscript:${cellData?.recipientLockscript} sudtExtraData:${cellData?.asset} sender:${cellData?.senderAddress}, confirmedNumber: ${confirmedNumber}, confirmed: ${confirmed}`,
+      );
+      logger.debug('CkbHandler watchLockEvents eth lockEvtLog:', { tx, cellData });
+      const bridgeFee = BigInt(ForceBridgeCore.config.eth.lockNervosAssetFee).toString();
+      await this.db.createCkbLock([
+        {
+          ckbTxHash,
+          chain,
+          senderAddress,
+          assetIdent: asset,
+          amount: `0x${amount.toString(16)}`,
+          bridgeFee,
+          recipientAddress,
+          blockNumber,
+          blockTimestamp: Number(block.header.timestamp),
+          confirmNumber: confirmedNumber,
+          confirmStatus,
+        },
+      ]);
+      BridgeMetricSingleton.getInstance(this.role).addBridgeTxMetrics('ckb_lock', 'success');
+      BridgeMetricSingleton.getInstance(this.role).addBridgeTokenMetrics('ckb_lock', [
+        {
+          amount: Number(amount),
+          token: asset,
+        },
+      ]);
+      logger.info(`CkbHandler watchLockEvents save CkbLock successful for ckb tx ${txInfo.tx.transaction.hash}.`);
+    }
+    if (records.length === 1) {
+      await this.db.updateLockConfirmNumber([{ ckbTxHash, confirmedNumber, confirmStatus }]);
+      logger.info(`update lock record ${txHash} status, confirmed number: ${confirmedNumber}, status: ${confirmed}`);
+    }
+    if (confirmed && this.role === 'collector') {
```

Comment:
should check before create mint record, ref: https://github.com/nervosnetwork/force-bridge/blob/a85d9b8e4ea4e4d55b55f2f86e22380e496e1f04/offchain-modules/packages/x/src/handlers/eth.ts#L224

Reply:
fixed, added checkLock and moved white list check, minimal bridge amount check, sudt size check into checkLock